src/network/taraxa_capability.cpp

```diff
@@ -159,8 +161,9 @@ void TaraxaCapability::onConnect(weak_ptr<Session> session, u256 const &) {
     cnt_received_messages_[_nodeID] = 0;
     test_sums_[_nodeID] = 0;
 
-    insertPeer(_nodeID);
+    auto peer = insertPeer(_nodeID);
     sendStatus(_nodeID, true);
+    peer->sent_initial_status_ = true;
```

Comment:
let's completely get rid of this `peer->sent_initial_status_`. If we call `sendStatus` before `insertPeer` we are sure it is set